app/controllers/companies_controller.rb

```diff
@@ -0,0 +1,42 @@
+class CompaniesController < ApplicationController
+  before_action :set_company, only: [:show, :update, :destroy]
+
+  def index
+    @companies = authorize Company.order(:name)
+    render json: @companies, status: :ok
+  end
+
+  def show
```

Comment:
Here we need to call `authorize`, don't we?
```suggestion
  def show
  authorize @company
```

Reply:
Not really. We have an action declared at the top of controller doing that.